packages/api/src/routes/medical/document.ts

```diff
@@ -108,29 +119,117 @@ router.post(
   })
 );
 
-/** ---------------------------------------------------------------------------
- * GET /downloadUrl
+/**
+ * Handles the logic for download url endpoints.
  *
+ * @param req Request object.
+ * @returns URL for downloading the document.
+ */
+async function getDownloadUrl(req: Request): Promise<string> {
+  const cxId = getCxIdOrFail(req);
+
+  const fileName = getFromQueryOrFail("fileName", req);
+  const type = getFrom("query").optional("conversionType", req);
+  const conversionType = type ? docConversionTypeSchema.parse(type) : undefined;
+
+  if ((typeof fileName !== "string" || fileName.indexOf(cxId) !== -1) && !Config.isSandbox()) {
+    throw new ForbiddenError();
+  }
+
+  const url = await downloadDocument({ fileName, conversionType });
+  return url;
+}
+
+// TODO: Redirect this endpoint to the new one (download-url)
+/** ---------------------------------------------------------------------------
+ * GET /document/downloadUrl
  * Fetches the document from S3 and sends a presigned URL
+ * @deprecated Use the GET /download-url endpoint instead.
  *
  * @param req.query.fileName The file name of the document in s3.
  * @param req.query.conversionType The doc type to convert to.
  * @return presigned url
  */
 router.get(
   "/downloadUrl",
+  asyncHandler(async (req: Request, res: Response) => {
+    const url = getDownloadUrl(req);
+    return res.status(MOVED_PERMANENTLY).json({ url });
```

Comment:
I think this is not correct, is it? When the server responds w/ 301 it indicates the URL the client should request next, not a logic related response payload (in this case the download URL).

We could just set status OK and leave it deprecated for now to simplify this PR